4985_CommAudio/server.cpp

```diff
@@ -3,9 +3,159 @@
+--
+--  RETURNS:       void
+--
+--  DATE:          April 4, 2017
+--
+--  DESIGNER:      Alex Zielinski
+--
+--  PROGRAMMER:    Alex Zielinski
+--
+--  NOTES:
+--      Stops audio from playing
+---------------------------------------------------------------------------------------*/
+void stopAudio()
+{
+    // check if audio is playing
+    if (output->state() == QAudio::ActiveState)
+    {
+        output->stop(); // stop the audio
+        output->reset();
+        audioBuffer->close();
+    }
+}
+
+/*--------------------------------------------------------------------------------------
+--  INTERFACE:     void pauseAudio()
+--
+--  RETURNS:       void
+--
+--  DATE:          April 4, 2017
+--
+--  DESIGNER:      Alex Zielinski
+--
+--  PROGRAMMER:    Alex Zielinski
+--
+--  NOTES:
+--      Suspend audio
+---------------------------------------------------------------------------------------*/
+void pauseAudio()
+{
+    // check if audio is playing
+    if (output->state() == QAudio::ActiveState)
+    {
+        output->suspend(); // pause the audio
+    }
+}
+
+/*--------------------------------------------------------------------------------------
+--  INTERFACE:     void playAudio(QString &filePath)
+--                      QString &filePath: the filepath of audio file
+--
+--  RETURNS:       void
+--
+--  DATE:          April 4, 2017
+--
+--  DESIGNER:      Alex Zielinski
+--
+--  PROGRAMMER:    Alex Zielinski
+--
+--  NOTES:
+--      initializes audio format and audio output
+---------------------------------------------------------------------------------------*/
+void playAudio(QString &filePath)
+{
+    // check if audio was paused
+    if(output->state() == QAudio::SuspendedState)
+    {
+        output->resume();
+        //qDebug() << audioBuffer->size();
+    }
+    else // audio not paused
+    {
+        // create file handle for audio file
+        QFile audioFile(filePath);
+
+        // open audio file
+        if (audioFile.open(QIODevice::ReadOnly))
+        {   // check if user selected a different track
+            if (audioFilePath != filePath)
+            {
+                audioFilePath = filePath;
+
+                // seek to raw audio data of wav file
+                audioFile.seek(AUDIODATA);
+
+                // extract raw audio data
+                QByteArray audio = audioFile.readAll();
+
+                // initialize audio buffer
+                audioBuffer = new QBuffer(&audio);
+                audioBuffer->open(QIODevice::ReadWrite);
+                audioBuffer->seek(0);
+                //qDebug() << audioBuffer->size();
+
+                output->start(audioBuffer); // play track
+
+                // event loop for tracck
+                QEventLoop loop;
+                QObject::connect(output, SIGNAL(stateChanged(QAudio::State)), &loop, SLOT(quit()));
+                do
+                {
+                    loop.exec();
+                } while(output->state() == QAudio::ActiveState);
```

Comment:
This isn't something you necessarily have to change, but I'm just curious as to why you're using do - while instead of just a while loop?